Installation fails on debian

Hello,

I am trying to `sudo make install` and I get the following error:
```
install -m644 doc/man/*.3 /usr/local/share/man/man3/
install: target '/usr/local/share/man/man3/' is not a directory: No such file or directory
```
Running `ls -lah /usr/local/share/man/` yields:
```
drwxrwsr-x  4 root staff 4.0K Mar 25 16:22 .
drwxrwsr-x 12 root staff 4.0K Jan  2 00:47 ..
drwxr-sr-x  2 root staff 4.0K Mar 25 16:34 man1
drwxr-sr-x  2 root staff 4.0K Mar 25 16:13 man{1,3,7}
```
If I replace the `install-man` entry in the makefile
with a dummy entry like `echo "hello"` I instead get
the following error message
```
install -m755 bin/lfe{,c,doc,script} /usr/local/lib/lfe/bin
install: cannot stat 'bin/lfe{,c,doc,script}': No such file or directory
Makefile:118: recipe for target 'install' failed
make: *** [install] Error 1
```

Any clues as to what I'm doing wrong? Help greatly appreciated.

--- a/Makefile
+++ b/Makefile
@@ -32,6 +32,8 @@ DESTBINDIR := $(DESTLIBDIR)/$(BINDIR)
 
 VPATH = $(SRCDIR)
 
+MKDIR_P = mkdir -p
+
 ERLCFLAGS = -W1
 ERLC = erlc
 
@@ -72,7 +74,7 @@ $(BINDIR)/%: $(CSRCDIR)/%.c
 	cc -o $@ $<
 
 $(EBINDIR)/%.beam: $(SRCDIR)/%.erl
-	@mkdir -p $(EBINDIR)
+	@$(MKDIR_P) $(EBINDIR)
 	$(ERLC) -I $(INCDIR) -o $(EBINDIR) $(COMP_OPTS) $(ERLCFLAGS) $<
 
 %.erl: %.xrl
@@ -214,7 +216,7 @@ $(DOCDIR)/%.txt: $(MANDIR)/%.7
 	groff -t -e -mandoc -Tutf8 -Kutf8 $< | col -bx > $@
 
 $(PDFDIR):
-	@mkdir -p $(PDFDIR)
+	@$(MKDIR_P) $(PDFDIR)
 
 docs-pdf: $(PDFDIR) \
 	$(addprefix $(PDFDIR)/, $(PDF1S)) \
@@ -231,7 +233,7 @@ $(PDFDIR)/%.pdf: $(DOCSRC)/%.7.md
 	pandoc -f markdown --latex-engine=xelatex -o $@ $<
 
 $(EPUBDIR):
-	@mkdir -p $(EPUBDIR)
+	@$(MKDIR_P) $(EPUBDIR)
 
 docs-epub: $(EPUBDIR) \
 	$(addprefix $(EPUBDIR)/, $(EPUB1S)) \
@@ -247,8 +249,10 @@ $(EPUBDIR)/%.epub: $(DOCSRC)/%.3.md
 $(EPUBDIR)/%.epub: $(DOCSRC)/%.7.md
 	pandoc -f markdown -t epub -o $@ $<
 
-install-man:
-	$(INSTALL_DIR) $(MANINSTDIR)/man{1,3,7}
+$(MANINSTDIR)/man%:
+	mkdir -p $@
+
+install-man: $(MANINSTDIR)/man1 $(MANINSTDIR)/man3 $(MANINSTDIR)/man7
 	$(INSTALL_DATA) $(MANDIR)/*.1 $(MANINSTDIR)/man1/
 	$(INSTALL_DATA) $(MANDIR)/*.3 $(MANINSTDIR)/man3/
 	$(INSTALL_DATA) $(MANDIR)/*.7 $(MANINSTDIR)/man7/

--- a/Makefile
+++ b/Makefile
@@ -116,6 +116,9 @@ comp_opts.mk:
 
 -include comp_opts.mk
 
+$(BINDIR)/lfe%:
+	$(INSTALL_BIN) $@ $(DESTBINDIR)
+
 install: compile install-man
 	rm -Rf $(DESTEBINDIR)
 	$(INSTALL_DIR) $(DESTEBINDIR)
@@ -123,7 +126,7 @@ install: compile install-man
 	$(INSTALL_DATA) $(addprefix $(EBINDIR)/, $(EBINS)) $(DESTEBINDIR)
 	$(INSTALL_DATA) $(addprefix $(EBINDIR)/, $(LBINS)) $(DESTEBINDIR)
 	$(INSTALL_DIR) $(DESTBINDIR)
-	$(INSTALL_BIN) $(BINDIR)/lfe{,c,doc,script} $(DESTBINDIR)
+	$(MAKE) $(BINDIR)/lfe $(BINDIR)/lfec $(BINDIR)/lfedoc $(BINDIR)/lfescript
 	ln -sf $(DESTBINDIR)/* $(PREFIX)/bin/
 
 clean:

--- a/Makefile
+++ b/Makefile
@@ -33,6 +33,7 @@ DESTBINDIR := $(DESTLIBDIR)/$(BINDIR)
 VPATH = $(SRCDIR)
 
 MKDIR_P = mkdir -p
+MANDB = $(shell which mandb)
 
 ERLCFLAGS = -W1
 ERLC = erlc
@@ -88,7 +89,7 @@ $(EBINDIR)/%.beam: $(LSRCDIR)/%.lfe
 
 all: compile
 
-.PHONY: compile erlc-compile lfec-compile erlc-lfec emacs install docs clean docker-build docker-push docker
+.PHONY: compile erlc-compile lfec-compile erlc-lfec emacs install docs clean docker-build docker-push docker update-mandb
 
 compile: comp_opts.mk
 	$(MAKE) $(MFLAGS) erlc-lfec
@@ -253,12 +254,20 @@ $(EPUBDIR)/%.epub: $(DOCSRC)/%.7.md
 	pandoc -f markdown -t epub -o $@ $<
 
 $(MANINSTDIR)/man%:
-	mkdir -p $@
+	@$(MKDIR_P) -p $@
 
+ifeq (,$(findstring mandb,$(MANDB)))
 install-man: $(MANINSTDIR)/man1 $(MANINSTDIR)/man3 $(MANINSTDIR)/man7
+else
+install-man: $(MANINSTDIR)/man1 $(MANINSTDIR)/man3 $(MANINSTDIR)/man7 update-mandb
+endif
 	$(INSTALL_DATA) $(MANDIR)/*.1 $(MANINSTDIR)/man1/
 	$(INSTALL_DATA) $(MANDIR)/*.3 $(MANINSTDIR)/man3/
 	$(INSTALL_DATA) $(MANDIR)/*.7 $(MANINSTDIR)/man7/
+
+update-mandb:
+	@echo "Updating man page database ..."
+	$(MANDB) $(MANINSTDIR)
 
 # Targets for working with Docker
 docker-build:

